### ISSUE TITLE
Can't connect to SAP R/3

### ISSUE THREAD

**Issue by charneca** [goenning/SharpSapRfc#41]
Hello, 

i'm trying to extract data from SAP R/3 However when it goes trough Method 
`public IRfcFunction CreateFunction(string name);`
I always get the following error:
An unhandled exception of type 'SharpSapRfc.SharpRfcCallException' occurred in SharpSapRfc.Plain.dll
Additional information: Function metadata lookup: unsupported backend version: -1

I can use older version of SAP connector not supported in this project, and I notice it have this property on older .net connector versions, not on NCO3
`        [Browsable(true)]
        [Category("General")]
        [DefaultValue("3")]
        [Description("RFC server type, 2/3/E/F: R/2 or R/3, External System or FastRFC (Default: 3)")]
        [SAPName("TYPE")]
        [TypeConverter(typeof(DestinationTypeConverter))]
        public virtual string Type { get; set; }`

So can anyone help me to connect to R/3 with SharpSapRfc ? Great project by the way, it's very hard to find SAP information on the web.

**Comment by goenning** [comment]
Hi,

Could you try to do the same connection with SAPNCo 3 without SharpSapRfc? I pretty sure this issue is not directly related to SharpSapRfc.

Which version/release your R/3 is running on? The message is not clear, but seems like this is something on your SAP instance.

**Comment by charneca** [comment]
Yes it's related to the SAPNco 3 that's why I would like to know if there is some way or configuration to connect to SAP R/3.  If i'm not wrong the release is 31I .

I can connect using another project that is using the old assembly "SAP.Connector.Rfc" 2.0.0.0 that is targeting .net 1.1 .  It also needs librfc32.dll.
The issue here is that the old assembly only compiles in 32 bits.

**Comment by docelmo** [comment]
ptnull, 

 Have you been able to find a resolution for this issue? My company uses SAP R/3 and has no plans to upgrade but like you I do not want to be stuck with a legacy library with no path to upgrade. 

Brian



> Yes it's related to the SAPNco 3 that's why I would like to know if there is some way or configuration to connect to SAP R/3. If i'm not wrong the release is 31I .
> 
> I can connect using another project that is using the old assembly "SAP.Connector.Rfc" 2.0.0.0 that is targeting .net 1.1 . It also needs librfc32.dll.
> The issue here is that the old assembly only compiles in 32 bits.

**Comment by charneca** [comment]
Hello, if I remember correctly I ended up creating a separate project library DLL with this assembly. 
My main 64bit project is consuming that when ir needs to conect to older versions of SAP.

**Comment by bgaboleiro** [comment]
> ptnull,
> 
> Have you been able to find a resolution for this issue? My company uses SAP R/3 and has no plans to upgrade but like you I do not want to be stuck with a legacy library with no path to upgrade.
> 
> Brian
> 
> > Yes it's related to the SAPNco 3 that's why I would like to know if there is some way or configuration to connect to SAP R/3. If i'm not wrong the release is 31I .
> > I can connect using another project that is using the old assembly "SAP.Connector.Rfc" 2.0.0.0 that is targeting .net 1.1 . It also needs librfc32.dll.
> > The issue here is that the old assembly only compiles in 32 bits.

Hello, if I remember correctly I ended up creating a separate project library DLL with this assembly.
My main 64bit project is consuming that when it needs to conect to older versions of SAP.
